src/hooks/useTracking.js

```diff
@@ -63,8 +63,14 @@ export const useTracking = (
   pageName,
   namespace = -1,
   sectionCount = 0,
-  openedSections = 0
+  openedSections = {}
 ) => {
+  const trackingRef = useRef()
+
+  useEffect(() => {
+    trackingRef.openedSectionCount = Object.keys(openedSections).length
+  }, [openedSections])
+
```

Comment:
Note :

We cannot simply use `openedSections` here, the reason is given here https://github.com/facebook/react/issues/14965

the suggestion is using `useRef` to handle things like mutable value
https://reactjs.org/docs/hooks-reference.html#useref